spec/unit/lib/wifi_user/use_cases/parse_email_request_spec.rb

```diff
@@ -0,0 +1,21 @@
+describe WifiUser::UseCase::ParseEmailRequest do
+  describe 'parsing the request' do
+    it 'returns the type when one is present' do
+      request = '{
+        "Type":"Notification",
+        "Message":"{\"mail\":{\"messageId\":\"123\",\"commonHeaders\":{\"from\":[\"bob@gov.uk\"],\"to\":[\"sally@gov.uk\"]}},\"receipt\":{\"action\":{\"objectKey\":\"some-object-key\",\"bucketName\":\"some-bucket-name\"}}}"
+      }'
```

Comment:
Realise this is pretty awful to read but this is the input to our system.
Not sure if converting this to and from something more readable is an unneeded layer of indirection.

Reply:
agree, let's stick with this. If it turns out modifying this is a pain point, we can change to encoding a plain object.

Reply:
Oof, yeah, probably doesn't actually help much to introduce the indirection, especially as we don't repeat this pattern (afaik).

This is a fairly small spec file - maybe it looks better spaced out a bit?

e.g.
```
"Message":"{
  \"mail\":{
    \"messageId\":\"123\",
    \"commonHeaders\":{
      \"from\":[\"bob@gov.uk\"],
      \"to\":[\"sally@gov.uk\"]}},
      \"receipt\":{
        \"action\":{
          \"objectKey\":\"some-object-key\",\"bucketName\":\"some-bucket-name\"
        }
      }
    }"
```

... still kind of horrible

Reply:
unfortunately you can't do that, as you cu'an't have multiple lines in JSON.
You'd have to escape the line endings too